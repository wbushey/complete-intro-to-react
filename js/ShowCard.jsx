// @flow
import React from 'react';
import styled from 'styled-components';
import {Link} from 'react-router-dom';

const Wrapper = styled.div`
  width: 32%;
  border: 2px solid #333;
  border-radius: 4px;
  margin-bottom: 25px;
  padding-right: 10px;
  overflow: hidden;
  text-decoration: none;
  color: black
`;

const Image = styled.img`
  width: 46%;
  float: left;
  margin-right: 10px;
`;

const ShowCard = (props: Show) => (
  <Link to={`/details/${props.imdbID}`}>
    <Wrapper>
      <Image
        alt={`${props.title} show Poster`}
        src={`/public/img/posters/${props.poster}`}
      />
      <h3>
        {props.title}
      </h3>
      <h4>
        ({props.year})
      </h4>
      <p>
        {props.description}
      </p>
    </Wrapper>
  </Link>
);

export default ShowCard;
