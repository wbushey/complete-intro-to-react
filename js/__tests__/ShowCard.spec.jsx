import React from 'react';
import {shallow} from 'enzyme';
import ShowCard from '../ShowCard';

test('ShowCard renders correctly', () => {
  const component = shallow(
    <ShowCard
      poster="foobar"
      title="Foobar"
      year="1990"
      description="A bar of foo"
    />,
  );
  expect(component).toMatchSnapshot();
});
