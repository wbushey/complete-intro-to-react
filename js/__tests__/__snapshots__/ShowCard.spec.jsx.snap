// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowCard renders correctly 1`] = `
<Link
  replace={false}
  to="/details/undefined"
>
  <styled.div>
    <styled.img
      alt="Foobar show Poster"
      src="/public/img/posters/foobar"
    />
    <h3>
      Foobar
    </h3>
    <h4>
      (
      1990
      )
    </h4>
    <p>
      A bar of foo
    </p>
  </styled.div>
</Link>
`;
